101 decoupled action spaces

AhmadAmine998 opened the pull request.

hzheng40 commented on gym/f110_gym/envs/action.py:
```diff
@@ -19,73 +19,134 @@ def from_string(action: str):
-
-        self.steering_low, self.steering_high = params["s_min"], params["s_max"]
-        self.velocity_low, self.velocity_high = params["v_min"], params["v_max"]
+        self._type = "speed"
```
typo? also should add steering so it's specified

hzheng40 commented on gym/f110_gym/envs/action.py:
```diff
@@ -19,73 +19,134 @@ def from_string(action: str):
+class SteeringSpeedAction(SteerAction):
+    def __init__(self, params: Dict) -> None:
+        super().__init__()
+        self._type = "speed"
```
add steering so it's specified

hzheng40 commented on gym/f110_gym/envs/action.py:
```diff
@@ -19,73 +19,134 @@ def from_string(action: str):
+
+    @staticmethod
+    def from_string(action: str):
+        if action == "angle":
```
again, should specifiy steering_angle and steering_speed, just to be less confusing

hzheng40 commented:
@AhmadAmine998 I pulled in the updated tests and it's failing right now. You'll probably have to update the tests to use the decoupled action spaces.
Also, should maybe consider handling when only one action type is set, (throw a warning and setting the other to default).

AhmadAmine998 commented:
> @AhmadAmine998 I pulled in the updated tests and it's failing right now. You'll probably have to update the tests to use the decoupled action spaces. Also, should maybe consider handling when only one action type is set, (throw a warning and setting the other to default).

I updated the initialization of the action space to be compatible with specifying only one control input as a string. The old tests passed and I now updated the tests to use the decoupled action specification. This way it is compatible with legacy code while still having updated tests

hzheng40 reviewed (approved):
Looks good.
